org.lflang/src/org/lflang/generator/Validator.java

```diff
@@ -97,7 +97,7 @@ public final void doValidate(CancelIndicator cancelIndicator) throws ExecutionEx
      * given by {@code getBuildReportingStrategies}, and return its return code.
      */
     public final int run(LFCommand command, CancelIndicator cancelIndicator) {
-        final int returnCode = command.run(cancelIndicator, true);
```

Comment:
> Maybe what we need is not a `--quiet` flag, but rather something like `--raw-errors`. So we could by default parse and process error messages from the target compiler, but give the user an option to see the raw error messages if needed.

This has been the design I have had in mind all along; report errors on LF code lines (also in `lfc`), but _also_ print the original error message (right after it) to provide context. I think that doing this by default makes sense, and a `--quiet` or `--hide-raw-errors` flag could be used to suppress the messages relating directly to the generated code.

Reply:
I guess the problem is, that with cargo this is not easily possible. Cargo is called with `--message-format json` to produce machine-readable error messages. But then they are not human readable. We would either need to run cargo twice, or parse the human readable messages and not the json output.

Reply:
The message should be the same, only the files/line number would be different, no?

Reply:
> This has been the design I have had in mind all along; report errors on LF code lines (also in lfc), but also print the original error message (right after it) to provide context.

One tricky aspect of this is that the errors we want to report differ according to whether we are in standalone mode. The whole message looks like this:
```
warning: unneeded `return` statement
 --> Preamble/src/reactors/preamble.rs:8:5
  |
8 |     return i + 42;
  |     ^^^^^^^^^^^^^^ help: remove `return`: `i + 42`
  |
  = note: `#[warn(clippy::needless_return)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
```
In the editor (but I assume not in standalone mode), we only want to report the first line ("unneeded return statement"). Everything else is more or less redundant for the language server because the marker appears in the editor. However, in standalone, this human readable output could be really helpful.

The JSON output has a field that specifies the human readable output which we could forward to stderr from the validator. We cannot change the line numbers in the human readable output (which is just a raw string) without doing something dirty, but we can supplement the string with the LF line number. I'll look into it in the next few days.

Reply:
> ```
> warning: unneeded `return` statement
>  --> Preamble/src/reactors/preamble.rs:8:5
>   |
> 8 |     return i + 42;
>   |     ^^^^^^^^^^^^^^ help: remove `return`: `i + 42`
>   |
>   = note: `#[warn(clippy::needless_return)]` on by default
>   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
> ```
> 
> In the editor (but I assume not in standalone mode), we only want to report the first line ("unneeded return statement"). 

I think that we would want to:
1) Report the message on the first line;
2) Report it in the right location inf the `.lf` file; and
3) Show the origin of the error/warning as it relates to the target code.

Something like:

 ```
 warning: unneeded `return` statement
  --> Foo.lf:3:5
   |
 3 |     return i + 42;
   |     ^^^^^^^^^^^^^^ help: remove `return`: `i + 42`
   |
  = origin: Preamble/src/reactors/bar.rs:8:5
  = note: `#[warn(clippy::needless_return)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
```

If we show it like this, then I think there's nothing redundant, and we even don't need a `--quiet` flag.